### ISSUE TITLE
An unbound VGG loss silently assumes a [0,1] output range

### ISSUE THREAD

**Issue by YeapJieShen** [YeapJieShen/single-image-super-resolution#239]
## The defect

`_VGGFeatureLoss.__init__` sets

```python
self._gain, self._offset = 1.0, 0.0
```

and `bind(processor)` later replaces them from `processor.output_range`. So a loss that was never
bound **silently behaves as though the model emits `[0, 1]`**, rather than failing.

Verified:

```
forward WITHOUT bind() -> succeeded, value 0.1684
_gain=1.0 _offset=0.0
```

## Why the default is the wrong choice here

The identical question was already answered the other way, one seam over.
`SRProcessor.output_range` is **abstract rather than defaulted**, and its docstring says why:

> Abstract, never defaulted: a wrong inherited default is the failure this exists to prevent.
> Reconstructing `[-1, 1]`-trained weights with `[0, 1]` logic produces silently wrong images and
> no error.

`SRLoss.bind` is abstract for the same stated reason. Yet the consumer of that carefully-abstract
value gives itself the identity as a fallback — which makes **"never bound"** and **"bound to a
`[0, 1]` processor"** indistinguishable at runtime.

## When it bites

Not through any shipped config: `SRLightning` binds an `SRLoss` criterion, and
`WeightedSumLoss.bind` forwards to its `SRLoss` terms, so both live paths bind.

It bites when a VGG term is reached through a container that is **not** itself an `SRLoss` — a
plain `nn.ModuleList`, or a user's own composite. Paired with `RGBSignedOutputProcessor`, whose
range is `[-1, 1]`, the loss then feeds `[-1, 1]` data through ImageNet normalisation designed for
`[0, 1]`. Training proceeds, the number looks plausible, and the features mean something other
than what the recipe says.

That is the flagship reproduction's content loss.

## Proposal

Make "never bound" loud, matching the seam's own stated reasoning: initialise to a sentinel and
raise in `_features` if it is still there, naming `bind` and the two live paths that call it.

The error should say what to do — put the term inside `WeightedSumLoss`, or call `bind` — because
the person who hits this has written a custom composite and needs to know the contract exists.

## Acceptance

- [ ] An unbound `_VGGFeatureLoss.forward` raises, naming `bind`
- [ ] Every shipped path is unaffected: direct `SRLoss` criterion, and nested in `WeightedSumLoss`
- [ ] A test proving the unbound call raises, RED before the change

**Comment by YeapJieShen** [comment]
**Fairness correction to the report above.** The default is deliberate, not an oversight, and the
code says so:

```python
# Identity until bind(); a [0, 1] processor needs no mapping anyway.
self._gain, self._offset = 1.0, 0.0
```

That reasoning is sound for the case it names — a `[0, 1]` processor genuinely needs no mapping,
so the identity is the correct *post-bind* value there.

The argument in this issue is narrower than "the default is wrong": it is that the same value
serves two different states, **"bound to a `[0, 1]` processor"** and **"never bound at all"**, and
only the first is the one the comment is reasoning about. A sentinel would keep the first
behaviour exactly and make the second loud.

So this is a request to distinguish two states, not to reverse a decision.
